File: problems/002/002.lua
posix = require "posix"

local function solution()
  local sum, curr, next_ = 0, 1, 2

  while curr < 4000000 do
    if curr % 2 == 0 then
      sum = sum + curr
    end

    local temp = next_
    next_ = next_ + curr
    curr = temp
  end

  return sum
end

if #arg == 1 and arg[1] == "-a" then
  print(solution())
  return
end

io.stdout:setvbuf("line")
for line in io.lines() do
  local iters = tonumber(line)

  local start = { posix.clock_gettime("monotonic") }
  for i = 1,iters do
    solution()
  end
  local end_ = { posix.clock_gettime("monotonic") }

  local secs = end_[1] - start[1]
  local nsecs = end_[2] - start[2]
  print(secs * 1000000000 + nsecs)
end

File: problems/002/002.js
function solution() {
    var sum = 0, curr = 2, next = 3;
    var temp;

    while (curr < 4000000) {
        if (curr % 2 === 0) {
            sum += curr;
        }

        temp = next;
        next += curr;
        curr = temp;
    }

    return sum;
}

if (process.argv.length === 3 && process.argv[2] === "-a") {
    console.log(solution());
    process.exit();
}

var clock = require("posix-clock");
var lazy = require("lazy");

new lazy(process.stdin).
    lines.
    forEach(function(line) {
        var iters = parseInt(line, 10);

        var start = clock.gettime(clock.MONOTONIC);
        var i;
        for (i = 0; i < iters; i++) {
            solution();
        }
        var end = clock.gettime(clock.MONOTONIC);

        var secs = end.sec - start.sec;
        var nsecs = end.nsec - start.nsec;

        console.log(secs * 1000000000 + nsecs);
    });
